content/en/blog/2022-03-25-putting-the-git-into-gitops/index.md

```diff
@@ -0,0 +1,212 @@
+The newly added support for commit signing with SSH keys is also not
+supported by our implementations at present.
+
+While the above might sound like petty implementation details, we had to
+learn ourselves that each Git implementation has its own set of
+shortcomings. Things which "just work" in the Git CLI, any of the
+implementations get subtly wrong, as they work on the ["plumbing"
+level](https://git-scm.com/book/en/v2/Git-Internals-Plumbing-and-Porcelain)
+of Git. In the end we chose to make `gitImplementation` a [configurable
+setting](/docs/components/source/gitrepositories/#git-implementation).
+
+Just to illustrate what happens when you try to do things just right,
+here's a couple of pieces of work we needed to get done along
+the way:
+
+- we had to add support for e.g. verifying `known_hosts` files for SSH
+  connections
+- when connecting over SSH, we received SHA1 and MD5 fingerprints of
+  the returned public key from the server and not the key itself,
+  which made `known_hosts` verification a little harder
+- changes on `libgit2` would [break the way that known hosts used to
+  work](https://github.com/fluxcd/source-controller/commit/9479d04779ccb7fc44b972cde23cb9a6c052f445)
+- Making various kinds of SSH key types work, e.g. support for
+  `ECDSA*` added as part of `libgit2 >= 1.2.0`, `ED25519` as part of
+  `libgit2 >= 1.3.0`.
+- we needed to start verifying PGP signatures
+
+## Tracking upstream developments in Git
+
+As Git has become ubiquitous and virtually all of the world's software
+development depends on Git, it still is in active development. Constant
+improvements affect features, efficiency, configurability and better
+security. Of course we want to pass this all down to our users: more
+efficient download makes a huge difference, support for Git Submodules
+enables new use-cases, support for more GPG verification or new SSH key
+formats adds additional security and when new features are rolled out in
+Git providers, we need to support them in Flux too.
+
+Integrating these changes into Flux unfortunately isn't as easy as it
+sounds. One part of the dependency chain for
+[git2go](https://github.com/libgit2/git2go) goes:
+
+```mermaid
+flowchart LR
+libgit2 --> libssh2 --> OpenSSL
+```
+
+So that's [libgit2](https://libgit2.org/),
+[libssh2](https://libssh2.org) (to enable the SSH transport) and
+[OpenSSL](https://www.openssl.org/). As Linux vendors often take a very
+conservative approach to bringing new software releases to stable
+releases, we were unfortunately pushed to
+[building these dependencies ourselves](https://github.com/fluxcd/golang-with-libgit2#rationale).
+In addition to that, these libraries have quite a few configuration
+options that can only be set at build time and unfortunately the
+`openssl`/`libssh2` packages of different Linux distributions act in
+[slightly different
+ways](https://github.com/fluxcd/golang-with-libgit2/blob/main/hack/Makefile#L63-L69).
```

Comment:
A versioned link would be better here to avoid pointing to unrelated section in the future.
```suggestion
ways](https://github.com/fluxcd/golang-with-libgit2/blob/libgit2-1.3.0/hack/Makefile#L63-L69).
```